src/resources/embeddings.ts

```diff
@@ -9,9 +9,46 @@ export class Embeddings extends APIResource {
    */
   create(
     body: EmbeddingCreateParams,
-    options?: Core.RequestOptions,
+    options?: Core.RequestOptions<EmbeddingCreateParams>,
   ): Core.APIPromise<CreateEmbeddingResponse> {
-    return this._client.post('/embeddings', { body, ...options });
+    const userInitialEncodingFormat = body.encoding_format;
+    Core.debug('request', 'Sending request with arguments:', { body, ...options });
+
+    const base64Response = this._client.post<EmbeddingCreateParams, CreateEmbeddingResponse>('/embeddings', {
+      body: {
+        ...body,
+        // Force base64 encoding for vector embeddings creation
+        // See https://github.com/openai/openai-node/issues/1310
+        encoding_format: 'base64',
```

Comment:
I don't think we want to *always* use `base64`, if the user explicitly asked for a different format we should have the exact same behaviour as we do prior to this PR which is to just let them.

Reply:
yes I think the tradeoff of not getting the perf optimisation is worth it. in general we do not want to get in the users way, if the user explicitly requests `float` then we should just send `float`.

having a way to opt-out of certain SDK things can be very helpful, especially if say, a new encoding format is added, then users wouldn't be able to actually use that until we update this logic to support it. however if we just send custom formats as-is then it'll just work, no matter what SDK version they're on

Reply:
@RobertCraigie I updated the logic. We now default to base64 only if the user does not provide en encoding_format.

Reply:
@RobertCraigie I updated the logic based on your feedback. We now default to base64 _only_ if the user does not provide en encoding_format. Let me know if that works for you.

Reply:
@RobertCraigie Please let me know if there is anything else missing.